src/oss-find-squats-lib/Mutators/IMutator.cs

```diff
@@ -51,15 +51,15 @@ public IEnumerable<Mutation> Generate(PackageURL arg)
                 if (hasNamespace)
                 {
                     yield return new Mutation(
-                        mutated: arg.CreateWithNewNames(arg.Name, HttpUtility.UrlEncode(mutation.Mutated)).ToString(),
+                        mutated: arg.CreateWithNewNames(arg.Name, mutation.Mutated).ToString(),
```

Comment:
Is it safe to remove this URL encoding here? I assume we used to have this to ensure no unescaped potentail dangerous mutations made it out of this method before. What is the rationale for changing it now with this PR? Is there a change inthe packageurl project that is related to if this needs to be preencoded?

Reply:
@jpinz[ it looks like you authored the change introducing the original encoding calls](https://github.com/microsoft/OSSGadget/commit/96ac5f107a2d729bee7cd820bbdf353ee0ae5cb3) - would you be able to comment on this? Were these calls originally introduced to work around issues generating correctly-encoded package URLs?

@gfs Mounika has been doing some work upstream in the packageurl-dotnet library to improve how it handles special characters in package names, package versions, and other metadata. Specifically, the packageurl-dotnet library did not implement proper percent- encoding/decoding of special Package URL characters present in attributes of package versions (including `namespace`, `name`, `version`, etc.) as required by spec. The newer version of the packageurl-dotnet library brings the packageurl-dotnet library into compliance with the spec.

As a result, it is now correct to treat individual package URL components handled by the `PackageURL` class (e.g. `Namespace`, `Name`, `Version`, etc.) as always unencoded - the `PackageURL` class takes care of percent- encoding/decoding each component when generating/parsing package URLs. Users of the library no longer need to (and in fact _must_ no longer) pass around individual components of a package URL in encoded form. This makes sense because data encoding is always context-dependent - passing data around in encoded form away from the context in which the encoding matters just makes everyone's lives harder.

Does that help?

@morended I noticed that no description was submitted with this PR. As a general rule, it's wise to include a description with every PR, particularly when bumping a library version requires changing how certain data values are represented. It's important to empathize with code reviewers and give them as much up-front context as we can to help them understand why the changes we propose are appropriate.

Reply:
Apologies for not updating the description. Updated the description. @pmalmsten  Thanks for clarifying that. 
@jpinz  Let me know if you have any objections with this change.

Reply:
> @jpinz[ it looks like you authored the change introducing the original encoding calls](https://github.com/microsoft/OSSGadget/commit/96ac5f107a2d729bee7cd820bbdf353ee0ae5cb3) - would you be able to comment on this? Were these calls originally introduced to work around issues generating correctly-encoded package URLs?

I don't recall exactly, but I'm pretty sure I added the encoding just to prevent any problems that might occur if they weren't encoded.

Reply:
@pmalmsten it sounds like the sanitization is now handled in the dependency so I think this would be fine - but agree with the feedback about including an explanation for the rationale in the description.